nearcore/src/config.rs

```diff
@@ -1884,12 +1882,13 @@ mod tests {
         assert!(key == test_ok("key", "fred", ""));
         test_err("key", "barney", "");
 
-        // test_seed == Some → the same key is generated
+        // test_seed == Some → the same key is generated (same signature is produced)
         let k1 = test_ok("k1", "fred", "foo");
         let k2 = test_ok("k2", "barney", "foo");
         let k3 = test_ok("k3", "fred", "bar");
 
-        assert!(k1.public_key == k2.public_key && k1.secret_key == k2.secret_key);
+        let data: &[u8] = b"Example data for signature test";
+        assert!(k1.public_key() == k2.public_key() && k1.sign(&data) == k2.sign(&data));
```

Comment:
Just curious, why we changed from `k1.secret_key == k2.secret_key` to `k1.sign(&data) == k2.sign(&data)` ?

Reply:
The test here checked if given the same seeds we end-up with the same public and secret keys. Since here we changed from InMemorySigner to Signer we do not have a public members now and we need public getters. 
  
So I wasn't sure if giving a public access method to "secret_key" is the best way to go (especially when the only attempts to retrieve it were in tests) so other way to compare SecretKeys would be to sing something with them (as there is no random and one-shot nonce generated inside the SecretKey while signing we can expect the same result).